src/d3a/d3a_core/sim_results/device_statistics.py

```diff
@@ -62,28 +61,45 @@ def _device_price_stats(cls, area: Area, subdict: Dict):
             if t.seller == area.name or t.buyer == area.name:
                 trade_price_list.append(t.offer.energy_rate / 100.0)
 
-        if trade_price_list != []:
-            _create_or_append_dict(subdict, key_name, {market.time_slot: trade_price_list})
+        if trade_price_list:
+            _create_or_append_dict(subdict, key_name,
+                                   {market.time_slot: trade_price_list})
         else:
             _create_or_append_dict(subdict, key_name, {market.time_slot: FILL_VALUE})
 
         cls._calc_min_max_from_sim_dict(subdict, key_name)
 
     @classmethod
     def _device_energy_stats(cls, area: Area, subdict: Dict):
-        key_name = "trade_energy_kWh"
         market = list(area.parent.past_markets)[-1]
-        traded_energy = 0
-        for t in market.trades:
-            if t.seller == area.name:
-                traded_energy -= t.offer.energy
-            if t.buyer == area.name:
-                traded_energy += t.offer.energy
-
-        _create_or_append_dict(subdict, key_name,
-                               {market.time_slot: traded_energy})
+        from d3a.models.strategy.infinite_bus import InfiniteBusStrategy
+        if type(area.strategy) == InfiniteBusStrategy:
```

Comment:
I think it is better if you create 2 functions, one for the if block and one for the else block. Then it will be clear which function is for calculating the statistics of the Infinite Bus, and which function is for the rest of the devices.